### DISCUSSION TITLE
Features planned to be paywalled

### DISCUSSION THREAD

**Discussion by neliets** [InvoiceShelf/InvoiceShelf#245] [Q&A] [answered]
Hello!

First of all - big thanks for the product. It is excellent and honestly only user-friendly platform available anywhere. And absolutely amazing job with the updates and releases.

Since every project has the looming question of funding. Likely not going to be purely donation based, and you probably have already plans with the modules to have a "marketplace", or earn from potentially support contracts or perhaps SaaS product.

But my question is, in any of the scenarios, have you discussed what would be under paywall/license/enterprise-only/whatever you call it?

**Comment by rihards-simanovics** [comment]
This is a very good question!

As the OP pointed, it is very difficult for Open Source projects to consistently fund themselves without need for introduction of enterprise edition or any other paid option or marketplace.

As of yet I am not aware of any no plans for any changes, however, and this is a speculation, if we were I would assume the first step would likely be the introduction of sponsorship on GitHub followed by a popup in the UI (similar to thunderbird or Coolify).

I can't say if there is going to be any more but the two above if implemented most likely will come with an option to disable them. As for the marketplace there is likely going to be a mix of both paid and free plugins, some official, some third party, likely similar to WordPress with a one click install.

@gdarko what are your thoughts?

**Comment by gdarko** [comment, marked as answer]
Hello,

First and foremost, we will keep **existing** features as they are, free.

**Regarding Modules:**

In the near future, we plan to introduce a modules directory, which will initially host only free modules. Over time, we may introduce paid modules, which will be licensed under GPLv3. These modules will interact with the backend and frontend through a clearly separated and isolated integration layer that relies solely on native Laravel and JS/Vue APIs for licensing purposes.

The long-term goal is to rewrite the code and eventually transition the entire project’s license from AGPL to GPLv3.

**Regarding Funding:**

Currently all the time that we put on the project is self-sponsored. There are several other OSS contributors currently that are helping with PRs and support voluntarily too. To support users who prefer a hassle-free experience, I considered offering a "set-it-and-forget-it" hosting service. This service may provide a simple, UI-based setup for creating one or multiple instances of InvoiceShelf, ideal for those who don’t want to manage the app on their own.

The cost of such service will be symbolic—comparable to the price of a few cups of coffee per month, depending on the plan. This will help cover infrastructure costs and possibly sponsor additional development time to sustain and improve the project. 

An added benefit should be reduction of support tickets related to the app set up because we currently receive high volume of support inquiries from users struggling with installation. While i've made an efforts to simplify the process with Docker, unfortunately, many users have limited knowledge in Docker.

Best regards,
Darko
